src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2798,6 +2793,39 @@ private void setEHcount(uint cEH)
 
         private void setEHinfo(uint EHnumber, ref CORINFO_EH_CLAUSE clause)
         {
+            // Filters don't have class token in the clause.ClassTokenOrOffset
+            if ((clause.Flags & CORINFO_EH_CLAUSE_FLAGS.CORINFO_EH_CLAUSE_FILTER) == 0)
+            {
+                if (clause.ClassTokenOrOffset != 0)
+                {
+                    MethodIL methodIL = _compilation.GetMethodIL(MethodBeingCompiled);
+                    mdToken classToken = (mdToken)clause.ClassTokenOrOffset;
+                    TypeDesc clauseType = (TypeDesc)ResolveTokenInScope(methodIL, MethodBeingCompiled, classToken);
+
+                    CORJIT_FLAGS flags = default(CORJIT_FLAGS);
+                    getJitFlags(ref flags, 0);
+
+                    if (flags.IsSet(CorJitFlag.CORJIT_FLAG_IL_STUB))
+                    {
+                        // IL stub tokens are 'private' and do not resolve correctly in their parent module's metadata.
+
+                        // Currently, the only place we are using a token here is for a COM-to-CLR exception-to-HRESULT
+                        // mapping catch clause.  We want this catch clause to catch all exceptions, so we override the
+                        // token to be mdTypeRefNil, which used by the EH system to mean catch(...)
+                        Debug.Assert(clauseType.IsObject);
+                        clause.ClassTokenOrOffset = 0;
+                    }
+                    else
+                    {
+                        // For all clause types add fixup to ensure the types are loaded before the code of the method
+                        // containing the catch blocks is executed. This ensures that a failure to load the types would
+                        // not happen when the exception handling is in progress and it is looking for a catch handler.
+                        // At that point, we could only fail fast.
+                        classMustBeLoadedBeforeCodeIsRun(clauseType);
```

Comment:
The clauseType can end up being `Object` here (for catch-all handlers). Since `Object` type is always loaded, we could skip adding the fixup here. But old crossgen generates the fixup for it too, so I have made it the same way here. @davidwrighton do you think we should bother excluding it?

Reply:
I don't understand your comment here. This is trying to protect the following situation...
When SomeFuncThatThrows throws its exception, the EH system needs to check and see if the catch clause matches SomeUnusedException. If we don't preload the SomeUnusedException type, then it will load that type as the EH processing occurs, which the comment here specifies isn't a safe time to load types. How does forcing clauseType to be Object in all cases help?
```
using System;

class SomeUnusedException : Exception {}

class Test
{
    static void SomeFuncThatThrows()
    {
        throw new Exception();
    }
    public static void Main(string[] args)
    {
        try
        {
            SomeFuncThatThrows();
        }
        catch (SomeUnusedException e)
        {
        }
    }
}
```

Reply:
I think you can test this by running the little sample I wrote up under a chk/dbg  build of the runtime. (In fact if it does repro we might want to add it to our test suite as a pri1 test.)

Reply:
Ah, I now see what you mean by Object for catch all handlers.. That's an interesting thought. I don't think that works, because there are 2 factors in play here.
1. The actual type being loaded.
2. The path through type-forwarders to the type being loaded. 

Both of those need to be handled. Our rule is that System.Object is always loaded, but it is possible (although … unlikely) that the EH clause could refer to System.Object through a type-forwarder assembly that isn't actually loaded. Also, I doubt that this would be a meaningful amount of optimization. There aren't actually that many global catch clauses in a typical application relative to the amount of code loaded.